src/record.rs

```diff
@@ -0,0 +1,296 @@
+use std::convert::TryFrom;
+use std::fmt::Debug;
+use std::sync::{Arc, Mutex};
+
+use crate::TermLike;
+
+/// A Terminal that records all writes and curor moves to it.
+/// Useful for debugging operations on terminals.
+#[cfg_attr(docsrs, doc(cfg(feature = "in_memory")))]
+#[derive(Debug, Clone)]
+pub struct RecordTerm {
+    state: Arc<Mutex<RecordTermState>>,
+}
+
+impl RecordTerm {
```

Comment:
In general, I wonder if it makes more sense to just merge the functionality of `RecordTerm` into `InMemoryTerm`?

Reply:
Yea, that makes sense. I mostly didn't know whether `InMemoryTerm` had any non-test usage